isis/src/tgo/apps/tgocassisstitch/tgocassisstitch.cpp

```diff
@@ -66,20 +66,69 @@ namespace Isis {
       throw IException(e, IException::Unknown, msg, _FILEINFO_);
     }
 
+    // read the optional cubename
+    FileName frameletCubeFlag(ui.GetCubeName("CUBENAME"));
+    QString frameletCubeName = frameletCubeFlag.expanded();
+
     // Stitch together the individual frames
-    FileName outputFileName(ui.GetCubeName("OUTPUTPREFIX"));
-    QString outputBaseName = outputFileName.expanded();
+    FileName outputPrefix(ui.GetCubeName("OUTPUTPREFIX"));
+    FileName outputSuffix(ui.GetCubeName("OUTPUTSUFFIX"));
+    QString outputPrefBaseName = outputPrefix.expanded();
+    QString outputSuffBaseName = outputSuffix.expanded();
     QStringList frameKeys = frameMap.uniqueKeys();
     Progress stitchProgress;
     stitchProgress.SetText("Stitching Frames");
     stitchProgress.SetMaximumSteps(frameKeys.size());
     stitchProgress.CheckStatus();
+    
     foreach(QString frameKey, frameKeys) {
       try {
         QString frameIdentifier = frameKey.split("/").last();
-        FileName frameFileName(outputBaseName + "-" + frameIdentifier + ".cub");
-        stitchFrame( frameMap.values(frameKey), frameFileName );
-        stitchProgress.CheckStatus();
+        if ((outputPrefix != "nil") && (outputSuffix == "nil")) {
```

Comment:
We should attempt to minimize the number of logical branches here.  Instead of having a branch for each combination of prefix / suffix, can we use something like:

```
if(outputPrefix == "nil"){
  outputPrefix = "";
}
if (outputSuffix == "nil"){
  outputSuffix = "";
}
```
Then we would just need conditionals for frameidentifier vs cubename, and we could get rid of all the nested logic e.g.:
```
FileName frameFileName(outputPrefBaseName + frameletCubeName + outputSuffBaseName ".cub");
```

I know this will get rid of the '-', but that's something that the user could easily control by specifying the separator as part of the prefix/suffix.  This might even be viewed as a "feature" since the user could then specify a '.' separator if they desire.  It's somewhat common to see something like mydata.proj.cub.